oauthenticator/google.py

```diff
@@ -373,13 +376,51 @@ def _fetch_user_groups(self, user_email, user_email_domain, http=None):
             credentials=credentials,
             http=http,
         )
+        return service
+
+    def _fetch_member_groups(
+        self,
+        member_email,
+        user_email_domain,
+        http=None,
+        checked_groups=set(),
+        processed_groups=set(),
```

Comment:
I think you may be required to have this be `..._groups=None`, because I think otherwise its a risk that the default value is re-used when it shouldn't be.

Googling made me find [this response](https://www.reddit.com/r/learnpython/comments/118ucmz/why_do_python_functions_reuse_pointer_to_mutable/#:~:text=Functions%20are%20first%20class%20objects,uses%20that%20same%20default%20argument.:):

>Functions are first class objects in python, and the default argument is set during the function definition, not when its called.
>
>So the empty list is getting assigned as the default argument on function definition.

So I think indeed, this has to be assigned `None`, and if/when needed, initialized within the function to an empty set.